sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/EmbeddingItem.java

```diff
@@ -43,9 +44,10 @@ private EmbeddingItem(@JsonProperty(value = "embedding") List<Double> embedding,
      *
      * @return the embedding value.
      */
-    @Generated
-    public List<Double> getEmbedding() {
-        return this.embedding;
+    public List<Float> getEmbedding() {
+        List<Float> floatList = new ArrayList<>();
+        this.embedding.forEach(d -> floatList.add(d.floatValue()));
```

Comment:
We should do this only once and not each time `getEmbedding()` method is called.

Reply:
https://github.com/Azure/azure-sdk-for-java/pull/39296/files#diff-8bbc3696d2aa378e509328482aceb1a89c1526b7c58e666407b0da901edcaff2R52